src/procedures/__tests__/PushDividendPayment.ts

```diff
@@ -0,0 +1,312 @@
+    contextMock.set('factories', factoriesMockedSetup);
+  });
+
+  afterEach(() => {
+    restore();
+  });
+
+  describe('Types', () => {
+    test('should extend procedure and have PushDividendPayment type', async () => {
+      target = new PushDividendPayment(params, contextMock.getMockInstance());
+      expect(target instanceof Procedure).toBe(true);
+      expect(target.type).toBe(ProcedureType.PushDividendPayment);
+    });
+  });
+
+  describe('PushDividendPayment', () => {
+    test('should throw if there is no valid security token being provided', async () => {
+      // Instantiate PushDividendPayment with incorrect security symbol
+      target = new PushDividendPayment(params, contextMock.getMockInstance());
+
+      tokenFactoryMock
+        .mock('getSecurityTokenInstanceFromTicker')
+        .withArgs(params.symbol)
+        .throws();
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: `There is no Security Token with symbol ${params.symbol}`,
+        })
+      );
+    });
+
+    test('should throw if there is no valid dividend type being provided', async () => {
+      // Instantiate PushDividendPayment with incorrect dividend type
+      target = new PushDividendPayment(
+        { ...params, dividendType: 'wrong' as DividendType },
+        contextMock.getMockInstance()
+      );
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should throw if an Erc20 dividend module is not attached', async () => {
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Erc20 },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock('getAttachedModules', Promise.resolve([]));
+
+      // Real call
+      await expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should throw if an Eth dividend module is not attached', async () => {
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Eth },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock('getAttachedModules', Promise.resolve([]));
+
+      // Real call
+      await expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should add a transaction to push a dividend payment of an attached ERC20 dividends module', async () => {
```

Comment:
Maybe, "should add a transaction to push a dividend payment *for* an attached ERC20 dividends module"